ragna/deploy/_ui/components/metadata_filters_builder.py

```diff
@@ -0,0 +1,220 @@
+import panel as pn
+import param
+
+from ragna.core._metadata_filter import MetadataFilter, MetadataOperator
+
+EXCLUDED_OPERATORS = ["RAW", "AND", "OR"]
+ALLOWED_OPERATORS = [
+    op
+    for op in list(MetadataOperator.__members__.keys())
+    if op not in EXCLUDED_OPERATORS
+]
+
+NO_CORPUS_KEY = "No corpuses available"
+NO_FILTER_KEY = "Empty Filter"
+
+
+class FilterRow(pn.viewable.Viewer):
+    key = param.Selector(objects=[NO_FILTER_KEY], default=NO_FILTER_KEY)
+    operator = param.Selector(objects=[""], default="")
+    value = param.Selector(objects=[""], default="")
+    multi_value = param.ListSelector(objects=[""], default=[])
+
+    def __init__(self, on_delete_callback, key_value_pairs=None, **params):
+        super().__init__(**params)
+
+        if key_value_pairs:
+            self.key_value_pairs = key_value_pairs
+            self.param.key.objects.extend(list(key_value_pairs.keys()))
+            key_select_disabled = False
+        else:
+            key_select_disabled = True
+
+        self.key_select = pn.widgets.Select.from_param(
+            self.param.key,
+            name="",
+            css_classes=["metadata-filter-key"],
+            disabled=key_select_disabled,
+        )
+
+        self.operator_select = pn.widgets.Select.from_param(
+            self.param.operator,
+            name="",
+            css_classes=["metadata-filter-operator"],
+            disabled=True,
+        )
+
+        self.value_select = pn.widgets.Select.from_param(
+            self.param.value,
+            name="",
+            css_classes=["metadata-filter-value"],
+            disabled=True,
+        )
+
+        self.multi_value_select = pn.widgets.MultiChoice.from_param(
```

Comment:
1. I like the idea and the widget in general, but right now I'm only able to see a single row 

   ![image](https://github.com/user-attachments/assets/1bad3f05-8ac1-4848-93c0-fec0aafc1bbb)

   Plus, it also leads to another vertical scroll bar so we now have two scroll bars on a modal, which is a big no-no according to Smera.
2. Not sure where this is handled, but the value is extracted wrong from the widget. For example if I select the following filter in the modal

   ![image](https://github.com/user-attachments/assets/c4b79ce0-c66a-4572-8be4-c9c5275ec0c7)

   I get the following output in the chat history

   ![image](https://github.com/user-attachments/assets/aa298a10-ea05-4b53-a4e9-d990428b1efa)

   There are two issues here:

   1. It did pick a different document as I did select. In my test I couldn't make out a pattern other than it only ever picked a single one.
   2. The value is still a string rather than a list of strings. This leads to errors in the backed, e.g. Chroma, as reported in a previous round of review.